src/EnergyPlus/WeatherManager.cc

```diff
@@ -3556,128 +3557,49 @@ namespace WeatherManager {
             PresWeathCodes = "999999999";
         }
         current_line.remove_prefix(pos + 1);
-        pos = index(current_line, ',');
-        if (pos != std::string::npos) {
-            if (pos != 0) {
-                bool error = false;
-                PrecipWater = UtilityRoutines::ProcessNumber(current_line.substr(0, pos), error);
-                if (error) {
-                    ErrorInterpretWeatherDataLine(state, WYear, WMonth, WDay, WHour, WMinute, Line, current_line);
-                }
-            } else {
-                PrecipWater = 999.0;
+
+        auto readNextNumber =
```

Comment:
Reusable lambda is nice here.  It's a little difficult to parse as a diff, but the result is much nicer.